chat_shell/chat_shell/services/context.py

```
@@ -0,0 +1,524 @@
+                "[CHAT_CONTEXT] Connected %d MCP servers: %s",
+                len(mcp_summary),
+                ", ".join(mcp_summary),
+            )
+
+        return mcp_tools, mcp_clients
+
+    async def _close_mcp_client(self, client) -> None:
+        """Close a single MCP client safely."""
+        try:
+            await client.close()
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)